### PR TITLE
Added option to generate.sh to use custom kubecon 2018 containers

### PR DESCRIPTION
**Pull request by sflxn**
Convenience argument to generate.sh to update the generated provider-components.yaml
file with the containers used during kubecon 2018.

### CODE REVIEW

File: cmd/clusterctl/examples/vsphere/generate-yaml.sh [hunk 1]
```diff
@@ -106,6 +116,23 @@ kustomize build $VSPHERE_CLUSTER_API_CRD_PATH/default/ > $PROVIDERCOMPONENT_GENE
 echo "---" >> $PROVIDERCOMPONENT_GENERATED_FILE
 kustomize build $CLUSTER_API_CRD_PATH/default/ >> $PROVIDERCOMPONENT_GENERATED_FILE
 
+if [ $USE_KUBECON = 1 ]; then
+  echo "Updating generated provider-components.yaml file with containers from kubecon 2018..."
+#  CONTROLLER_CONTAINER=$(grep cluster-api-controller ${PROVIDERCOMPONENT_GENERATED_FILE})
+  CONTROLLER_CONTAINER=`grep "image:.*cluster-api-controller" ${PROVIDERCOMPONENT_GENERATED_FILE} | awk '{ print $2 }'`
+  VSPHERE_PROVIDER_CONTAINER=`grep "image:.*vsphere-cluster-api" ${PROVIDERCOMPONENT_GENERATED_FILE} | awk '{ print $2 }'`
+
+  cat $PROVIDERCOMPONENT_GENERATED_FILE | \
```
Comment (dougm) [inline cmd/clusterctl/examples/vsphere/generate-yaml.sh]: sed supports multiple `-e` args.  You can also use `-i` to edit the file in place:

```sh
sed -i "$PROVIDERCOMPONENT_GENERATED_FILE" \
    -e "s|$VSPHERE_PROVIDER_CONTAINER|$KUBECON_VSPHERE_PROVIDER|" \
    -e "s|$CONTROLLER_CONTAINER|$KUBECON_CONTROLLER|"
```

File: cmd/clusterctl/examples/vsphere/generate-yaml.sh [hunk 2]
```diff
@@ -52,6 +57,7 @@ while test $# -gt 0; do
             echo "options:"
             echo "-h, --help                show brief help"
             echo "-f, --force-overwrite     if file to be generated already exists, force script to overwrite it"
+            echo "--kubecon					Use the containers from the kubecon demo"
```
Comment (dougm) [inline cmd/clusterctl/examples/vsphere/generate-yaml.sh]: cosmetic: misaligned with the other help and the other help text starts with lower-case.

File: cmd/clusterctl/examples/vsphere/generate-yaml.sh (hunk 1 above)
Comment (sidharthsurana) [inline cmd/clusterctl/examples/vsphere/generate-yaml.sh]: line#125-133 can be re-written in a single command
`sed -i -e "s|$VSPHERE_PROVIDER_CONTAINER|$KUBECON_VSPHERE_PROVIDER|" -e "s|$CONTROLLER_CONTAINER|$KUBECON_CONTROLLER|" $PROVIDERCOMPONENT_GENERATED_FILE`

Comment (sflxn) [reply cmd/clusterctl/examples/vsphere/generate-yaml.sh]: -i doesn't work on Mac unless you install gnu sed.

File: cmd/clusterctl/examples/vsphere/generate-yaml.sh (hunk 2 above)
Comment (sflxn) [reply cmd/clusterctl/examples/vsphere/generate-yaml.sh]: I'll fix that in a future PR

File: cmd/clusterctl/examples/vsphere/generate-yaml.sh (hunk 1 above)
Comment (sflxn) [reply cmd/clusterctl/examples/vsphere/generate-yaml.sh]: I answered doug above around this topic.

**Pull request merged**